.github/workflows/release.yml

```diff
@@ -0,0 +1,108 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      publish-tag:
+        description: 'The tag of the version to publish'
+        required: true
+        type: string
+
+concurrency:
+  group: release
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_TERM_COLOR: always
+
+jobs:
+  test-release:
+    name: Check & Test release
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - os: windows-latest
+        - os: ubuntu-latest
+        - os: macos-latest
+        - os: ubuntu-latest
+          env:
+            CARGO_BUILD_TARGET: wasm32-wasi
+            CARGO_TARGET_WASM32_WASI_RUNNER: /home/runner/.wasmtime/bin/wasmtime --dir=.
+    runs-on: ${{ matrix.os }}
+    if: github.ref == 'refs/heads/main'
+    env: ${{ matrix.env || fromJSON('{}') }}
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+
+    - name: Install Wasm deps
+      if: matrix.env.CARGO_BUILD_TARGET == 'wasm32-wasi'
+      run: |
+        rustup target add wasm32-wasi
+        curl -LO https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-20/wasi-sdk_20.0_amd64.deb
+        sudo dpkg --install wasi-sdk_20.0_amd64.deb
+        curl https://wasmtime.dev/install.sh -sSf | bash
+
+    - uses: Swatinem/rust-cache@v2
+      with:
+        shared-key: release
+        save-if: ${{ github.ref_name == 'main' }}
+
+    - run: rustup show
+
+    - name: Install cargo-hack
+      uses: taiki-e/install-action@cargo-hack
+
+    - name: Clippy
+      run: cargo hack clippy --feature-powerset -- -D warnings
+
+    - name: Test
+      run: cargo hack test --feature-powerset
+
+    - name: Check Documentation
+      env:
+        RUSTDOCFLAGS: '-D warnings'
+      run: cargo hack doc --feature-powerset
+
+    - name: Check semver
+      if: matrix.os == 'ubuntu-latest'
+      uses: obi1kenobi/cargo-semver-checks-action@v2
+
+  publish-release:
+    name: Publish release
+    needs: test-release
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        persist-credentials: true
+
+    - name: Install Rust toolchain
+      uses: moonrepo/setup-rust@v1
```

Comment:
Can you replace this with a more widely known package?

Reply:
Which one should I use ? I know this one from Biome CI/CD.

Reply:
rustup is available on the github runners. you can look into the existing CI file. I don't think you need this at all.

Reply:
Alright i'll remove it